### PR TITLE
MachineLICM: Allow hoisting REG_SEQUENCE

### PR DESCRIPTION
**Pull request by arsenm**
This is just a fancy copy. Extend the copy handling to cover
reg_sequence with only virtual register uses. This avoids some
test regressions in a future commit.

### CODE REVIEW

File: llvm/test/CodeGen/Hexagon/expand-vstorerw-undef.ll [hunk 1]
```diff
@@ -69,6 +69,7 @@ b18:                                              ; preds = %b16, %b7
   br label %b22
 
 b21:                                              ; preds = %b22
+  store volatile <64 x i32> %v20, ptr null
```
Comment (qcolombet) [inline llvm/test/CodeGen/Hexagon/expand-vstorerw-undef.ll:72]: Unrelated change?

**Review by qcolombet — APPROVED** [review]
LGTM.

Is the change to the Hexagon input expected, though?

**Comment by arsenm** [comment]
> LGTM.
> 
> 
> 
> Is the change to the Hexagon input expected, though?

Yes, otherwise the spill it seemed to be looking for disappears so this defeats the optimization
